infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/K8sContainerResolver.java

```diff
@@ -113,24 +109,8 @@ private void provisionMemoryLimit(Container container, CheContainer cheContainer
         .collect(Collectors.toList());
   }
 
-  private String buildContainerName(String pluginName, String cheContainerName) {
-    if (pluginName == null) {
-      return cheContainerName.substring(
-          0, min(cheContainerName.length(), MAX_CONTAINER_NAME_LENGTH));
-    }
-    String preliminaryName = (pluginName + "-" + cheContainerName).toLowerCase();
-    if (preliminaryName.length() <= MAX_CONTAINER_NAME_LENGTH) {
-      return preliminaryName;
-    }
-    final String limitedContainerName =
-        cheContainerName.substring(0, min(cheContainerName.length(), 49));
-    return (pluginName.substring(
-                0,
-                min(
-                    pluginName.length(),
-                    MAX_CONTAINER_NAME_LENGTH - limitedContainerName.length() - 1))
-            + "-"
-            + limitedContainerName)
-        .toLowerCase();
+  private String buildContainerName(String cheContainerName) {
+    String uniqueName = NameGenerator.generate(cheContainerName, 3).toLowerCase();
+    return uniqueName.substring(0, min(uniqueName.length(), MAX_CONTAINER_NAME_LENGTH));
```

Comment:
It affects UX, does not it? Can we generate only if there are collisions? Or at least always generate container names but use the original container name for machine name and generate prefix if needed.

Reply:
I do not think that how container on a k8s cluster is named is important for UX.
I changed approach because now broker sends both plugin and container name (previously only container name) and for exec plugin name was too long for k8s to use it in annotations.
Since plugin name wasn't sent previously code that used it in name was unreachable. 
Adding search for name collision brings additional complexity and I don't think it is worth to implement it. But surely, you could add it after if it will affect UX for you